build.gradle

```diff
@@ -352,6 +352,32 @@ subprojects {
     }
   }
 
+  jacocoTestReport {
+    dependsOn test // tests are required to run before generating the report
+  }
+
+  afterEvaluate {
+    jacocoTestCoverageVerification {
+      dependsOn jacocoTestReport
+
+      violationRules {
+        rule {
+          def threshold = project.ext.has('jacocoCoverageThreshold') ?
+              project.ext.jacocoCoverageThreshold : 0.8
+
+          limit {
+            counter = 'LINE'
```

Comment:
I'd suggest to use branch coverage, as it more indicative and doesn't suffer from line inflation.

Reply:
Sure. Did some [research](https://stackoverflow.com/questions/37169154/is-branch-coverage-as-useful-as-line-coverage).
Turned out branch coverage is more reasonable than line coverage and requires less effort than compound condition coverage